language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/AddCheckCodeAction.java

```diff
@@ -63,8 +63,11 @@ public AddCheckCodeAction() {
         if (!DIAGNOSTIC_CODES.contains(diagnostic.diagnosticInfo().code())) {
             return Collections.emptyList();
         }
-        Node matchedNode = getMatchedNode(positionDetails.matchedNode());
-        if (matchedNode == null) {
+
+        //Check if there is a check expression already present.
+        ExpressionNodeResolver expressionResolver = new ExpressionNodeResolver();
```

Comment:
Shall we rename the `ExpressionNodeResolver` to a bit of a specific one? This is generic

Reply:
What do you recommend? Something like `MacthedExpressionNodeResolver`?

Reply:
+1